internal/cmd/cmd.go

```diff
@@ -47,6 +47,11 @@ func Main(rawArgs []string) int {
 
 	cfg, err := ext.conf.Load(ctx, parsedArgs)
 	if err != nil {
+		if cfg == nil {
```

Comment:
I don't think this is the right way to decide that there were no config files.

The function used here wasn't intentionally designed with behavior "it should return nil for config only if no config file exists", and from reading the code I can see at least one case where this isn't the way it works (in loadFromPath):

```golang
		if prefs[env.Prefix()] {
			return nil, fmt.Errorf("duplicate environment definitions for '%s'", env.Prefix())
		}
```

So right now it returns nil if config file doesn't exist, and also in some cases where config file isn't valid.

You'd have to at least fix the above case in order for this approach to work and make a nil return value consistently mean "no config file exists".  I'm not sure it's a good idea though - pretty sure the dominant idiom is that, if error is non-nil, then you are not supposed to even look at the other returned values.  It's especially weird to use the non-error values to figure out what kind of error occurred.  Shouldn't the error object carry that info?

Also, whether or not cfg is nil is a single bit of information, so it's possible to use it to handle a single type of error.  But as soon as the requirement arises to handle another kind of error in a custom way, the approach falls down and this code would have to be refactored.

I think the intended approach to handling specific error cases is something like https://www.digitalocean.com/community/tutorials/creating-custom-errors-in-go , so you'd do e.g.

```go
// Declare a specific error type for no config file
type MissingConfigFile struct {
  // config file paths we checked
  candidates []string
}

// This makes it implement 'error' interface
func (m *MissingConfigFile) Error() string {
  return fmt.Sprintf("no existing config file in: %s", strings.Join(m.candidates, ", "))
}

// Then in the side returning an error, we can do:
return nil, &MissingConfigFile{candidates: candidates}

// And in the side receiving the error, if we get an error and we want to know specifically whether
// the error is that there's no config file:
cfg, err := ext.conf.Load(...)
if err != nil {
  if _, ok := err.(*MissingConfigFile); ok {
    // It was a missing config file, so handle that case
  } else {
    // It was some other kind of error
  }
}
```

I suppose you could also just test if the error string contained certain patterns.  But seeing as the above is I think the expected approach for error handling which can scale as the codebase grows, it might be good for us to get experience with it.